speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -217,6 +309,93 @@ static void SaveSettings(void* /* dummy */)
 }
 
+/// @brief         Sets the task setting in the researcher menu
+/// @param p_tasks boolean array that defines which task should be set
+void SetTask(bool* p_tasks) 
+{
+    if (p_tasks[TASK_LANE_KEEPING]) {
+        m_task = TASK_LANE_KEEPING;
+        return;
+    }
+    if (p_tasks[TASK_SPEED_CONTROL]) {
+        m_task = TASK_SPEED_CONTROL;
+        return;
+    }
+    m_task = TASK_NO_TASK;
+}
+
+/// @brief                    Sets the intervention setting in the researcher menu
+/// @param p_interventionType boolean array that defines which intervention type should be set
+void SetInterventionType(bool* p_interventionType)
+{
+    if (p_interventionType[INTERVENTION_TYPE_NO_SIGNALS]) {
+        m_interventionType = INTERVENTION_TYPE_NO_SIGNALS;
+        return;
+    }
+    if (p_interventionType[INTERVENTION_TYPE_ONLY_SIGNALS]) {
+        m_interventionType = INTERVENTION_TYPE_ONLY_SIGNALS;
+        return;
+    }
+    if(p_interventionType[INTERVENTION_TYPE_SHARED_CONTROL]) {
+        m_interventionType = INTERVENTION_TYPE_SHARED_CONTROL;
+        return;
+    }
+    if (p_interventionType[INTERVENTION_TYPE_COMPLETE_TAKEOVER]) {
+        m_interventionType = INTERVENTION_TYPE_COMPLETE_TAKEOVER;
+        return;
+    }
+    return;
+}
+
+/// @brief         Initializes the menu setting from the ResearcherMenu.xml file
+/// @param p_param The configuration menu handle
+void InitializeSettings(void* p_param)
+{
+    // Retrieve all setting variables from the xml file
+    bool* checkboxTasks = new bool[3];
+    checkboxTasks[TASK_LANE_KEEPING]  = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_LANE_KEEPING, GFMNU_ATTR_CHECKED, NULL), true);
+    checkboxTasks[TASK_SPEED_CONTROL] = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_SPEED_CONTROL, GFMNU_ATTR_CHECKED, NULL), false);
+
+    bool checkboxIndicatorAuditory = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_INDICATOR_AUDITORY, GFMNU_ATTR_CHECKED, NULL), true);
+    bool checkboxIndicatorVisual   = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_INDICATOR_VISUAL, GFMNU_ATTR_CHECKED, NULL), true);
+    bool checkboxIndicatorTextual  = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_INDICATOR_TEXT, GFMNU_ATTR_CHECKED, NULL), true);
+
+
+    bool* checkboxInterventions = new bool[5];
+    checkboxInterventions[INTERVENTION_TYPE_NO_SIGNALS] = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_TYPE_NO_SIGNALS, GFMNU_ATTR_CHECKED, NULL), true);
+    checkboxInterventions[INTERVENTION_TYPE_ONLY_SIGNALS] = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_TYPE_ONLY_SIGNALS, GFMNU_ATTR_CHECKED, NULL), false);
+    checkboxInterventions[INTERVENTION_TYPE_SHARED_CONTROL] = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_TYPE_SHARED_CONTROL, GFMNU_ATTR_CHECKED, NULL), false);
+    checkboxInterventions[INTERVENTION_TYPE_COMPLETE_TAKEOVER] = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_TYPE_COMPLETE_TAKEOVER, GFMNU_ATTR_CHECKED, NULL), false);
```

Comment:
No need to put into an array an assign a value later, but this will get fixed with the radio buttons feature.